src/core/useSpriteLoader.tsx

```diff
@@ -26,355 +75,411 @@ export const checkIfFrameIsEmpty = (frameData: Uint8ClampedArray) => {
+        }
+      } else if (_spriteTexture) {
+        spriteDataRef.current = json
+        spriteDataRef.current.frames = parseFrames()
+
+        totalFrames.current = Array.isArray(json.frames) ? json.frames.length : Object.keys(json.frames).length
+        const { w, h } = getFirstFrame(json.frames).sourceSize
+        aspect = calculateAspectRatio(w, h, aspectFactor)
       }
 
-      return sprites
-    } else {
-      // we need to convert it into an array
-      let spritesArr: any[] = []
+      setSpriteData(spriteDataRef.current)
 
-      for (const key in data.frames) {
-        spritesArr.push(data.frames[key])
+      if ('encoding' in _spriteTexture) {
+        _spriteTexture.encoding = 3001 // sRGBEncoding
+      } else if ('colorSpace' in _spriteTexture) {
+        //@ts-ignore
+        _spriteTexture.colorSpace = THREE.SRGBColorSpace
       }
 
-      spritesArr = calculateScaleRatio(spritesArr)
+      setSpriteTexture(_spriteTexture)
+      setSpriteObj({
+        spriteTexture: _spriteTexture,
+        spriteData: spriteDataRef.current,
+        aspect: aspect,
+      })
+    },
+    [getRowsAndColumns, numberOfFrames, parseFrames, calculateAspectRatio]
+  )
 
-      return spritesArr
-    }
-  }
+  /**
+   *
+   */
+  const loadJsonAndTextureAndExecuteCallback = React.useCallback(
+    (jsonUrl: string, textureUrl: string, callback: (json: SpriteData, texture: THREE.Texture) => void): void => {
+      const jsonPromise = fetch(jsonUrl).then((response) => response.json())
+      const texturePromise = new Promise<THREE.Texture>((resolve) => {
+        textureLoader.load(textureUrl, resolve)
+      })
+
+      Promise.all([jsonPromise, texturePromise]).then((response) => {
+        callback(response[0], response[1])
+      })
+    },
+    [textureLoader]
+  )
+
+  const loadStandaloneSprite = React.useCallback(
+    (textureUrl?: string) => {
+      if (!textureUrl && !inputRef.current) {
+        throw new Error('Either textureUrl or input must be provided')
+      }
 
-  // calculate scale ratio for the frames
-  const calculateScaleRatio = (frames: any[]) => {
-    // Find the largest frame
-    let largestFrame: any = null
-    for (const frame of frames) {
-      const { w, h } = frame.frame
-      const area = w * h
-      if (!largestFrame || area > largestFrame.area) {
-        largestFrame = { ...frame.frame, area }
+      const validUrl = textureUrl ?? inputRef.current
+      if (!validUrl) {
+        throw new Error('A valid texture URL must be provided')
       }
-    }
 
-    // Set scaleRatio property on each frame
-    for (const frame of frames) {
-      const { w, h } = frame.frame
-      const area = w * h
-      if (area === largestFrame?.area) {
-        frame.scaleRatio = 1
+      new Promise<THREE.Texture>((resolve) => {
+        textureLoader.load(validUrl, resolve)
+      }).then((texture) => {
+        parseSpriteData(null, texture)
+      })

```

Comment:
you don't use the promise object you create, so why not just a callback:

```suggestion
      textureLoader.load(validUrl, (tex) => {parseSpriteData(null, tex)})
```

Reply:
fixed in the latest commit